pkg/sql/opt/xform/coster.go

```diff
@@ -134,13 +133,269 @@ const (
 	// If the final expression has this cost or larger, it means that there was no
 	// plan that could satisfy the hints.
 	hugeCost memo.Cost = 1e100
-
-	// Some benchmarks showed that some geo functions were atleast 10 times
-	// slower than some float functions, so this is a somewhat data-backed
-	// guess.
-	geoFnCost = cpuCostFactor * 10
 )
 
+// fnCost maps some functions to an execution cost. Currently this list
+// contains only st_* functions, including some we don't have implemented
+// yet. Although function costs differ based on the overload (due to
+// arguments), here we are using the minimum from similar functions based on
+// postgres' pg_proc table. At some later date we can choose to add costs for
+// other functions.
+//   select proname, min(procost) from pg_proc where proname like 'st\_%' and procost > 1 group by proname order by proname
+var fnCost = map[string]memo.Cost{
```

Comment:
i wonder if these should these live on the overload themselves at some point (or even now :P).

Reply:
we can apply a test to ensure they match with postgres, as we do with volatility right now: https://github.com/cockroachdb/cockroach/blob/51bea831714d1e4b4aba5c17dd016f08e88c077b/pkg/sql/sem/builtins/all_builtins_test.go#L47

Reply:
Yes moving them to the overloads would be awesome. But more work than I'm willing to automate right now. I'm kinda on the "no" side of ensuring they match postgres because ours might likely have different performance characteristics than postgres. This current list is designed to just be better than what came before, not necessarily great.